app/lib/actions/pre_upgrade_action.rb

```diff
@@ -31,6 +30,13 @@ def format_output(job_output)
                            .join('')
         JSON.parse(output)
       end
+
+      def correct_job?(job_invocation)
+        category = job_invocation.job_category == ::ForemanLeapp::JOB_CATEGORY
+        rex_feature = job_invocation.remote_execution_feature&.label == 'leapp_preupgrade'
+
+        category && rex_feature
```

Comment:
Since ruby does not evaluate things lazily you still try to check if the label is `leapp_preupgrade`, even if it is irrelevant because of the category.

Reply:
Okay, but I'm not sure about correct fix. Like this?
```
      def leapp_preupgrade_job?(job_invocation)
        return false unless job_invocation.job_category == ::ForemanLeapp::JOB_CATEGORY
        return job_invocation.remote_execution_feature&.label == 'leapp_preupgrade'
      end
```

Reply:
Is there anything wrong with just this?
```ruby
job_invocation.job_category == ::ForemanLeapp::JOB_CATEGORY &&
  job_invocation.remote_execution_feature&.label == 'leapp_preupgrade'
```

Reply:
Nope, just wasn't sure if it's enough to fix the issue.